collections-dictionary/CompareTwoDictionaries/CompareTwoDictionaries/DictionaryEqualityComparerBenchmark.cs

```diff
@@ -0,0 +1,35 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Order;
+
+namespace CompareTwoDictionaries;
+
+[MemoryDiagnoser]
+[Orderer(SummaryOrderPolicy.FastestToSlowest)]
+public class DictionaryEqualityComparerBenchmark
+{
+    private static readonly Dictionary<int, string> _dict1
+        = new()
+        {
+            {1, "Rosary Ogechi"},
+            {2, "Clare Chiamaka"},
+        };
+
+    private static readonly Dictionary<int, string> _dict2
+        = new()
+        {
+            {1, "Rosary Ogechi"},
+            {2, "Clare Chiamaka"},
+        };
```

Comment:
This is very small for a benchmark. We should probably be using a dictionary with around 1000 elements, especially since we are benchmarking an OrderBy LINQ method, which does a bunch of stuff behind the scenes to create a map of index values to iterate the collection in the specified order, which means that it is allocating arrays internally, which of course will add pressure to the GC, so to get a more "realistic" benchmark, we should be looking at a minimum of 1000 items in the dictionary. We can easily generate those in code though:
`Enumerable.Range(1, 1000).ToDictionary(k => k, v => v.ToString())`

This has the added advantage of making the string comparison in the value equality check a little more intensive. Since the above strings are compile time constants, they are reference equal and the string comparison method first checks for reference equality, so each of the values above pass the equality check very quickly.